### PR TITLE
[enh] Add Petalsearch Images engine

### PR DESCRIPTION
**Pull request by allendema**
## What does this PR do?

Add engine to get results from Petalsearch Images.

## Why is this change important?
More image engines to choose.

## How to test this PR locally?
Do a search with it, like so:

```!pti mark normand```

## Author's checklist
N/A

## Related issues
https://github.com/searxng/searxng/issues/1257

### CODE REVIEW

**Comment by tiekoetter** [comment]
@allendema Please reformat with `make format.python`

File: searx/settings.yml [hunk 1]
```diff
@@ -1720,6 +1720,20 @@ engines:
       require_api_key: false
       results: HTML
 
+  - name: petal images
```
Comment (tiekoetter) [inline searx/settings.yml]:
```suggestion
  - name: petalsearch images
```

Comment (tiekoetter) [reply searx/settings.yml]: For reference: https://github.com/searxng/searxng/blob/51ba817e06bb15ca1768010d6873d1d7bf48b0b6/searx/settings.yml#L1723

File: searx/settings.yml [hunk 2]
```diff
@@ -1720,6 +1720,20 @@ engines:
...
+    engine: petal_images
+    shortcut: pti
```
Comment (tiekoetter) [inline searx/settings.yml]:
```suggestion
    shortcut: ptsi
```

Comment (tiekoetter) [reply searx/settings.yml]: For reference: https://github.com/searxng/searxng/blob/51ba817e06bb15ca1768010d6873d1d7bf48b0b6/searx/settings.yml#L1724

**Review by tiekoetter — CHANGES_REQUESTED** [review]

**Review by tiekoetter — APPROVED** [review]
LGTM

File: searx/engines/petal_images.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+categories = ['images']
+paging = True
+safesearch = True
+
+
+safesearch_table = {0: 'off', 1: 'moderate', 2: 'on'}
+
+base_url = 'https://petalsearch.com/'
+search_string = 'search?{query}&channel=image&ps=50&pn={page}&region={lang}\
+    &ss_mode={safesearch}&ss_type=normal'
```
Comment (return42) [inline searx/engines/petal_images.py]: FYI: This will add 4 (white-) spaces to the URL / I fixed it.

File: searx/settings.yml [hunk 4]
```diff
@@ -1720,6 +1720,20 @@ engines:
...
+    paging: true
+    categories: images
+    disabled: true
+    timeout: 3.0
+    about:
+      website: https://petalsearch.com/
+      wikidata_id: Q104399280
+      use_official_api: false
+      require_api_key: false
+      results: JSON
```
Comment (return42) [inline searx/settings.yml]: FYI: I moved this to the python code.

File: searx/settings.yml [hunk 5]
```diff
@@ -1720,6 +1720,20 @@ engines:
       require_api_key: false
       results: HTML
 
+  - name: petalsearch images
+    engine: petal_images
+    shortcut: ptsi
+    paging: true
+    categories: images
```
Comment (return42) [inline searx/settings.yml]: FYI: this is already defined in the python code / I removed this lines.

**Review by return42 — COMMENTED** [review]
@allendema thanks a lot for adding petalsearch image engine .. its a very fast engine (google feels ponderous compared) .. I left notes about some nits (I fixed) .. I am going to merge :+1:

**Pull request merged**